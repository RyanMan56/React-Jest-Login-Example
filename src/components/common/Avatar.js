import React from 'react';
import styled from 'styled-components';
import md5 from 'js-md5';
import defaultImage from '../../assets/face.svg';

const Styles = {
  Wrapper: styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-end;
    width: 130px;
    height: 130px;
    border: 6px solid #ffffff;
    border-radius: 50%;
    background-color: #f48d04;
    margin: 0 auto;
    transform: translateY(-35px);
    margin-bottom: -5px;
    overflow: hidden;
  `,
  Image: styled.img.attrs(({ hash }) => ({
    src: hash ? `https://www.gravatar.com/avatar/${hash}?s=130` : defaultImage
  }))`

  `,
};

const Avatar = ({ email }) => {
  const hash = email
    ? md5(email.trim().toLowerCase())
    : null;
  return (
    <Styles.Wrapper>
      <Styles.Image hash={hash} alt="avatar" />
    </Styles.Wrapper>
  )
};

export default Avatar;