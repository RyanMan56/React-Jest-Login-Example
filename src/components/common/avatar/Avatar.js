import React from 'react';
import styled from 'styled-components';
import md5 from 'js-md5';
import defaultImage from '../../../assets/face.svg';

const Styles = {
  Wrapper: styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-end;
    width: ${({ size }) => size || 130}px;
    height: ${({ size }) => size || 130}px;
    border: 6px solid #ffffff;
    border-radius: 50%;
    background-color: #f48d04;
    margin: 0 auto;
    margin-bottom: -5px;
    overflow: hidden;
  `,
  Image: styled.img.attrs(({ hash, size }) => ({
    src: hash ? getAvatar({ hash, size }) : defaultImage
  }))``,
};

const getAvatar = ({ hash, size }) => {
  return `https://www.gravatar.com/avatar/${hash}?s=${size || 130}`;
}

export const hashEmail = (email) => {
  return email
    ? md5(email.trim().toLowerCase())
    : null;
}

const Avatar = (props) => {
  const { email, size } = props;
  const hash = hashEmail(email);
  return (
    <Styles.Wrapper {...props}>
      <Styles.Image id="avatar-image" hash={hash} size={size} alt="avatar" />
    </Styles.Wrapper>
  )
};

export default Avatar;