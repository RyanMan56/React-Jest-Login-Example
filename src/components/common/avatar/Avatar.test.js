import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Avatar, { hashEmail } from './Avatar';

it('renders without crashing', () => {  
  shallow(<Avatar email="test@test.com" />);
});

it('matches snapshot', () => {
  const avatar = renderer
    .create(
      <Avatar email="test@test.com" />
    ).toJSON();

  expect(avatar).toMatchSnapshot();
});

it('successfully hashes email address', () => {
  expect(hashEmail("test@test.com")).toBe('b642b4217b34b1e8d3bd915fc65c4452');
});
