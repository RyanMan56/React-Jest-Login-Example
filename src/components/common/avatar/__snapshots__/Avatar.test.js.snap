// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdVaJa pLCSA"
>
  <img
    alt="avatar"
    className="sc-bwzfXH uTxCW"
    id="avatar-image"
    src="https://www.gravatar.com/avatar/b642b4217b34b1e8d3bd915fc65c4452?s=130"
  />
</div>
`;
